### ISSUE TITLE
Bad interaction of indexing_slicing with windows/chunks_exact

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#5676]
By contract, windows and chunks_exact will always yield valid slices therefore the following should be allowed:

```rust
#![deny(clippy::indexing_slicing)]

fn main() {
    let bytes = [1, 2, 3, 4, 5];
    bytes.as_ref().windows(2).for_each(|x| { dbg!(x[0], x[1]); });
}
```

**Comment by flip1995** [comment]
`indexing_slicing` is a `restriction` lint, which is designed to _restrict_ the language. In this case, to disallow indexing or slicing, in any case. "False positives" like this are therefore expected and seen as a "special case" and therefore should be `#[allow]`ed, if verified by hand. (It's a feature, not a bug :wink:).

**Comment by c410-f3r** [comment]
@flip1995 Thanks for answering 

This reasoning could be applied to user functions but special care should be applied to std functions. Well, this is just me and since this is a feature, not much can be done.

**Comment by flip1995** [comment]
The thought behind restriction lints is, that they are only relevant for really specific code bases, where for example a panic because of indexing is never acceptable. If we would then implement some "hacks" to exclude cases like this, it is more likely, that there will be false negatives. This would be a bigger problem for folks who want to forbid indexing at all, than putting an `allow` on some carefully checked occurrences. So while for every other lint we're using FN>FP as a rule of thumb, for restriction lints we usually apply FP>FN.
